### PR TITLE
Add route endpoint to messages sent and received metrics

### PR DESCRIPTION
**Pull request by lfitchett**
Edge hub has the following message metrics: edgehub_messages_received_total and 
edgehub_messages_sent_total. We have customer feedback to add the **bold** tags to the metrics.

edgehub_messages_received_total tags:
iothub
edge_device
protocol
id
**route_output**

edgehub_messages_sent_total tags:
iothub
edge_device
from
**from_route_output**
to
**to_route_input**

### CODE REVIEW

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/MessageHelper.cs [hunk 1]
```diff
@@ -14,5 +15,25 @@ public static string GetSenderId(this IMessage message)
 
             return string.Empty;
         }
+
+        public static string GetOutput(this IMessage message)
```
Comment (varunpuranik) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/MessageHelper.cs]: Nit: GetOutput() => TryGetValue() ? outputName : string.Empty;

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/linkhandlers/ModuleMessageLinkHandler.cs [hunk 2]
```diff
@@ -45,13 +46,17 @@ static class Metrics
...
-                new List<string> { "protocol", "from", "to" });
+                new List<string> { "from", "to", "from_route_output", "to_route_input" });
 
             public static void AddMessage(IIdentity identity, IMessage message)
             {
                 string from = message.GetSenderId();
                 string to = identity.Id;
-                MessagesMeter.Increment(1, new[] { "amqp", from, to });
+
+                // Note these are switched, because the message's output is edgeHub's input and vice-versa
```
Comment (varunpuranik) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/linkhandlers/ModuleMessageLinkHandler.cs]: Emm.. this does not make sense. EdgeHub has no concepts of Input/outputs. Those are the module concepts.

Comment (lfitchett) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/linkhandlers/ModuleMessageLinkHandler.cs]: I meant it in terms of 'from' being edgeHub's input (the message's output tag) and 'to' being edgeHub's output (the message's input tag).

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/MessageHelper.cs (hunk 1 above)
Comment (lfitchett) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/MessageHelper.cs]: Does this improve readability? I find the of statement faster to parse visually

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/linkhandlers/ModuleMessageLinkHandler.cs (hunk 2 above)
Comment (lfitchett) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/linkhandlers/ModuleMessageLinkHandler.cs]: actually, looking at the actual prometheous tag it corresponds to, I'll rename my variables.

Comment (lfitchett) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/linkhandlers/ModuleMessageLinkHandler.cs]: Its now fromRouteOutput and toRouteInput

**Pull request merged**